api/app/Actions/EventType/CloneEventTypeByIdAction.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Actions\EventType;
+
+use App\Exceptions\EventType\NameIsAlreadyInUseException;
+use App\Exceptions\EventType\SlugIsAlreadyInUseException;
+use App\Repositories\EventType\Criterion\NameCriterion;
+use App\Repositories\EventType\Criterion\SlugCriterion;
+use App\Repositories\EventType\EventTypeRepositoryInterface;
+
+final class CloneEventTypeByIdAction
+{
+    private EventTypeRepositoryInterface $eventTypeRepository;
+    private const MAX_NAME_LENGTH = 50;
+
+    public function __construct(EventTypeRepositoryInterface $eventTypeRepository)
+    {
+        $this->eventTypeRepository = $eventTypeRepository;
+    }
+
+    public function execute(CloneEventTypeByIdRequest $request): CloneEventTypeByIdResponse
+    {
+        $eventType = $this->eventTypeRepository->getById($request->getId());
+        $newEventType = $eventType->replicate();
+
+        $newSlug = $newEventType->slug . '-clone';
```

Comment:
Can we add some random string to be sure it will be unique?

Reply:
If such slug already in database we will receive an error.

Reply:
Yes, we will receive an error instead of getting the event type copied.

Reply:
It is not really important, but better to have this working in such cases.